### PR TITLE
Nomination field values should be from previously saved nominations

### PR DESCRIPTION
**Pull request by youcatastrophe**

### CODE REVIEW

File: app/views/nominations/nomination_ballot.html.erb [hunk 1]
```diff
@@ -7,27 +7,39 @@
...
-        <% nomination_count += 1 %>
-      <% end %>
+
+<% if @users_prev_noms.count == 0 %>
+  <%= form_for :nomination, url: {controller: "nominations", action: "save_nominee"} do |n| %>
+    <% @awards.each do |a| %>
+      <%= a.award_name %><br>
+      <% nomination_count = 0 %>
+        <% while nomination_count <= 4 %>
+
```
Comment (sumeetjain) [inline app/views/nominations/nomination_ballot.html.erb]: Remove trailing whitespace

**Comment by halfghaninne** [comment]
Please let me know if this is helpful, from [RoR's documentation on Form Helpers](http://guides.rubyonrails.org/form_helpers.html)

"2 Dealing with Model Objects
2.1 Model Object Helpers
A particularly common task for a form is editing or creating a model object. While the *_tag helpers can certainly be used for this task they are somewhat verbose as for each tag you would have to ensure the correct parameter name is used and set the default value of the input appropriately. Rails provides helpers tailored to this task. These helpers lack the _tag suffix, for example text_field, text_area.

For these helpers the first argument is the name of an instance variable and the second is the name of a method (usually an attribute) to call on that object. Rails will set the value of the input control to the return value of that method for the object and set an appropriate input name. If your controller has defined @person and that person's name is Henry then a form containing:

`<%= text_field(:person, :name) %>`
will produce output similar to

`<input id="person_name" name="person[name]" type="text" value="Henry"/>` "

We can talk more about this if you want to work on this before Sumeet is next available. I'm not 100% across what you're trying to accomplish.

**Comment by youcatastrophe** [comment]
Thanks Alex! I've actually been researching that same documentation. So we're at least on the same page. Let me read through it a little more and try a few modifications of my current (awful) code. If Sumeet isn't available by that point, it would be great to talk it over with you 😀

**Comment by sumeetjain** [comment]
Let's use this as an example:

``` erb
<input type="text" name="awards[<%= a.id %>][nominations[<%= nomination_count %>][theater]]" value="FFFFFF">
```

We want FFFFFF to be the value of one of the current user's nomination for that award. Specifically we want it to be the value of their `nomination_count`th nomination for award `a`.

So write a method in the `User` model:

``` ruby
def nomination_value(award_id, number)
end
```

Have that method return the value you need it to.

Then change `<input type="text" name="awards[<%= a.id %>][nominations[<%= nomination_count %>][theater]]" value="FFFFFF">` to:

``` erb
<input type="text" name="awards[<%= a.id %>][nominations[<%= nomination_count %>][theater]]" value="<%= current_user.nomination_value(a.id, nomination_count) %>">
```

You don't even need the conditional that creates two identical forms where one has user values and the other does not. And this task doesn't need to have anything to do with `form_for` (although it _could_, but that's not a necessary step).

Always just start with: What HTML am I trying to generate? Okay, how do I make Ruby methods to get me the information I need.

**Comment by youcatastrophe** [comment]
I appreciate the help and critiques, Sumeet! Much obliged.

**Comment by youcatastrophe** [comment]
`def nomination_value(award_id, nom_count, value_needed)
    nominations = Nomination.where("award_id" => award_id, 
      "user_id" => self.id) 
    @nomination = nominations[nom_count]
    @nomination.value_needed
  end  
`

The above is the method I'm currently trying to write. It is _not_ working the way I want it to.

**Comment by halfghaninne** [comment]
What are you expecting it to return and what error (or incorrect information) are you receiving instead? Precision here will help me help you.

**Pull request merged**